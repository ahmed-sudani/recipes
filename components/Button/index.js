import styles from "./styles.module.css";
export default function Button(props) {
  return <button className={styles.btn}>{props.text}</button>;
}
