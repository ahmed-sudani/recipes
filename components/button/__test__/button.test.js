import { render, screen } from '@testing-library/react'
import Button from '..'

describe('Button', () => {
  it('should render button component', () => {
    render(<Button name="test" />)
    const buttonElement = screen.getByRole('button')
    expect(buttonElement).toMatchSnapshot()
  })
  it('should match the text passed to button', () => {
    render(<Button name="test" />)
    const buttonElement = screen.getByRole('button')
    expect(buttonElement.textContent).toMatch(/test/i)
  })
})
