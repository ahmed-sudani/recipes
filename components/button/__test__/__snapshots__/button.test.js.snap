// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should render button component 1`] = `
<button
  class="btn"
  name="test"
>
  test
</button>
`;
